lib/project_types/extension/features/argo_cli_compatibility.rb

```diff
@@ -0,0 +1,46 @@
+module Extension
+  module Features
+    class ArgoCliCompatibility
```

Comment:
The thought for this is that this would handle any sort of cli compatibility (`argo-admin-cli` or `argo-run`)

Reply:
@joeyfreund let's definitely set up a time to chat!

The quick summary is that:

`argo-admin-cli` is used by argo extensions that  use the `argo-admin-template`. `argo-run` is used by extensions  that use the `argo-checkout-template`.  So these are 2 different clis we need to support.